Makefile

```diff
@@ -17,14 +17,21 @@ LDFLAGS_COMMON := -X main.gitCommit=$(COMMIT) -X main.version=$(VERSION)
 
 GOARCH := $(shell $(GO) env GOARCH)
 
+# -trimpath may be required on some platforms to create reproducible builds
+# on the other hand, it does strip out build information, like -ldflags, which
+# some tools use to infer the version, in the absence of go information,
+# which happens when you use `go build`.
+# This enables someone to override by doing `make runc TRIMPATH= ` etc.
+TRIMPATH := -trimpath
```

Comment:
I think we might want to call this something like `BUILD_FLAGS_REPRODUCIBLE=` because `-buildvcs=false` is something we will also want to add, and it will also strip debug information that users might want (notably the commit id). I think it's fair to say that most of the build flags we will add for reproducibility are going to be related to removing debug information that is less reproducible...

Or maybe you don't care about the vcs information auto-added to Go binaries in the recent versions of Go?

Reply:
I can see use cases both ways. My immediate preference is a flag targeted on trimpath, but if you prefer the flag as you put it, I'm fine changing it. What's your preference?

Reply:
> BUILD_FLAGS_REPRODUCIBLE

Can we use SOURCE_DATE_EPOCH ?

Reply:
What does SDE have to do with whether or not we set trimpath?

Reply:
If it is set, it is expected to be reproducible

Reply:
Oh. If set, run -trimpath, else not? That does change the default from "no special var set = trimpath" to "only if special var set = trimpath", but if that's what you want, I can change this PR. Let me know.